### PR TITLE
Add sinks to tokamax splash

### PR DESCRIPTION
**Pull request by RissyRan**
# Description

Add sink feature to tokamax splash attention

# Tests

* Test it locally on [Cloud](https://screenshot.googleplex.com/C34hCajt86L4nfa), no obvious perf delta is observed 
* Expect tests are green in runners

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: src/MaxText/layers/attention_op.py [hunk 1]
```diff
@@ -1288,12 +1288,12 @@ def wrap_flash_attention(
...
         if max_logit_value is not None:
-          attention_output = jax.vmap(partial(splash_kernel, max_logit_value=max_logit_value))(
-              query, key, value, decoder_segment_ids_tuple
-          )
-        else:
-          attention_output = jax.vmap(splash_kernel)(query, key, value, decoder_segment_ids_tuple)
+          kernel = partial(splash_kernel, max_logit_value=max_logit_value)
+        attention_output = jax.vmap(lambda q, k, v, d, s: kernel(q, k, v, d, sinks=s), in_axes=(0, 0, 0, 0, None))(
+            query, key, value, decoder_segment_ids_tuple, sinks
+        )
```
[the remark this answers is not in the document]
Comment (RissyRan) [reply src/MaxText/layers/attention_op.py:1294]: I doubt we need to apply `sinks=` key word, as sinks is not directly after decoder_segment_ids_tuple in Tokamax API.

File: src/MaxText/layers/attention_op.py [hunk 2]
```diff
@@ -1291,12 +1288,12 @@ def wrap_flash_attention(
...
 
       if self.config.use_tokamax_splash:
+        kernel = splash_kernel
         if max_logit_value is not None:
-          attention_output = jax.vmap(partial(splash_kernel, max_logit_value=max_logit_value))(
-              query, key, value, decoder_segment_ids_tuple
-          )
-        else:
-          attention_output = jax.vmap(splash_kernel)(query, key, value, decoder_segment_ids_tuple)
+          kernel = partial(splash_kernel, max_logit_value=max_logit_value)
```
Comment (suexu1025) [inline src/MaxText/layers/attention_op.py]: can you move this to before max_logit_value?

Comment (RissyRan) [reply src/MaxText/layers/attention_op.py]: Updated.
